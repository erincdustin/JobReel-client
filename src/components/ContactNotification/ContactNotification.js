import React from 'react';
import jobReelApiService from '../../services/jobreel-api-service';
import jobReelContext from '../../context/JobReelContext';

class ContactNotification extends React.Component {

  static contextType = jobReelContext

  state = {
    contact_name: this.props.name,
    job_title: this.props.job_title,
    company: this.props.company,
    email: this.props.email,
    linkedin: this.props.linkedin,
    comments: this.props.comments,
    connected: this.props.connected,
    date_connected: this.props.date_connected,
    contact_id: this.props.contact_id,
    date_added: this.props.date_added,
    notification: this.props.notification,
    user_id: this.props.user_id
  }

  handleHideContact() {
    const { contact_name, job_title, company, email, linkedin, comments, connected, date_connected, contact_id, date_added, user_id } = this.state
    const editedContact = { 
      contact_name, 
      job_title, 
      company, 
      email, 
      linkedin,
      comments, 
      connected,
      date_connected,
      contact_id,
      date_added,
      notification: false,
      user_id
      }
    jobReelApiService.editContact(editedContact, contact_id)
    this.context.updateContact(editedContact)
    this.setState({ notification: false })
  }

  renderNotification(){
    let today = new Date();
    let connected = new Date(this.state.date_connected)
    let diff = Math.abs((today - connected) / (1000 * 60 * 60 * 24))
    let rounded = Math.round(diff)
    let mail=`mailto:${this.state.email}`
    if (this.state.date_connected && (this.state.notification === true) && (rounded > 7)) {
    return <p  
    className="notification"
    key={this.state.contact_id}>
      You reached out to {this.state.contact_name} {rounded} days ago.  
      <a href={mail}> Follow up </a>
       if you haven't received a response.
       <button className="notification-button" onClick={() => this.handleHideContact()}>Clear</button></p>
      }
  }
  
  render() {
    return (
      <div className="contact-notification">
        {this.renderNotification()}
      </div>
    )
  }
}

export default ContactNotification;