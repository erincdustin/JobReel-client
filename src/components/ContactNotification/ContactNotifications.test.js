import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ContactNotification from './ContactNotification'

const props = {
  contact_id: 1,
  contact_name: 'Contact 1',
  job_title: 'Engineer',
  company: 'Company1',
  email: 'email@email.com',
  linkedin: 'http://www.linkedin.com/person1',
  comments: 'Contact 1 comments',
  date_added: '2019-07-03T19:26:38.918Z',
  connected: false,
  user_id: 1,
  date_connected: '2019-08-03T19:26:38.918Z',
  notification: true
  }

describe('ContactNotification smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<ContactNotification {...props}/>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('ContactNotification snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<ContactNotification {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})