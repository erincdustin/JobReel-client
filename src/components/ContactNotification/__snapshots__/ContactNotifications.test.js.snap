// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactNotification snapshot test renders the UI as expected 1`] = `
<div
  className="contact-notification"
>
  <p
    key="1"
  >
    You reached out to 
     
    25
     days ago.  
    <a
      href="mailto:email@email.com"
    >
      Follow up
    </a>
     if you haven't received a response.
    <button
      onClick={[Function]}
    >
      Clear
    </button>
  </p>
</div>
`;
